geopyspark/geotrellis/catalog.py

```diff
@@ -423,19 +423,22 @@ def write(uri,
             be in camel case. If both options and keywords are set, then the options will be used.
     """
 
+    if not tiled_raster_layer.zoom_level:
+        raise ValueError("The given layer does not have a zoom_level", tiled_raster_layer)
```

Comment:
This is a problem. In GeoTrellis when we're writing a floating layer it gets the  zoom level of 0. In PySpark the `LocalLayers` layers have `zoom_level` of `None`.

Since we must be able to write a `LocalLayer`, we should use zoom level of 0. On the read side we when not given a zoom we should read the 0th level but report the `zoom_level` as `None`. I think this will keep a consistent view of what is going on.